source/algorithms.tex

```diff
@@ -4550,12 +4550,16 @@
 The type \tcode{Size}
 is convertible to an integral type\iref{conv.integral,class.conv}.
 
+\pnum
+Let $E$ be \tcode{pred(*(i + n), value)}
```

Comment:
We have `!= false` in the existing text. That does not matter if `pred` is required to return `bool`, but is it?

Reply:
It's required to model _`boolean-testable`_ which means that `!= false` is already wrong.

This is the subject of the recently opened [LWG 4127](https://cplusplus.github.io/LWG/issue4127). I can incorporate this change into the proposed resolution of that issue if you prefer.

Reply:
The LWG issue was only opened after I created this pull request (possibly as a result of seeing it). Since we now have an LWG issue touching that text, maybe we shouldn't change it editorially.

Reply:
side note: LWG4127 should use "is true" everywhere.

I think I'm good with applying the fix here editorially, although I'd prefer to retain the `!= false` in the editorial rewrite and fix that part in the LWG issue. But we don't have to burden the LWG issue with editorial rewrites such as this one.

Reply:
OK, I've pushed a change to restore the `!= false`